src/main/java/com/google/api/gax/grpc/ServiceApiSettings.java

```diff
@@ -92,12 +96,17 @@ public ScheduledExecutorService getExecutor() {
    *
    * See class documentation for more details on channels.
    */
-  public ServiceApiSettings provideChannelWith(final ManagedChannel channel) {
+  public ServiceApiSettings provideChannelWith(
+      final ManagedChannel channel, final boolean shouldAutoClose) {
```

Comment:
We have been trying to avoid the functionality of closing a channel that someone provides to us. Let's just assume that if it is provided, we don't own it and won't auto close it.